### PR TITLE
[CI] Add clang job and setup concurrency

### PR DESCRIPTION
**Pull request by christophfroehlich**
- This adds a semi-binary job using clang. I fixed all what clang was complaining about.
- use humble branch of realtime_tools for humble CI

Additionally, I added the concurrency settings which now automatically cancel running workflows if anything is pushed to PRs.
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs

### CODE REVIEW

File: ros2_controllers.jazzy.repos [hunk 1]
```diff
@@ -2,7 +2,7 @@ repositories:
   ros2_control:
     type: git
     url: https://github.com/ros-controls/ros2_control.git
-    version: master
+    version: ci/clang
```
Comment (christophfroehlich) [inline ros2_controllers.jazzy.repos:5]:
```suggestion
    version: master
```

File: ros2_controllers.rolling.repos [hunk 2]
```diff
@@ -2,7 +2,7 @@ repositories:
   ros2_control:
     type: git
     url: https://github.com/ros-controls/ros2_control.git
-    version: master
+    version: ci/clang
```
Comment (christophfroehlich) [inline ros2_controllers.rolling.repos:5]:
```suggestion
    version: master
```

File: ros2_controllers.jazzy.repos [hunk 3]
```diff
@@ -14,7 +14,7 @@ repositories:
   control_toolbox:
     type: git
     url: https://github.com/ros-controls/control_toolbox.git
-    version: ros2-master
+    version: ci/clang
```
Comment (christophfroehlich) [inline ros2_controllers.jazzy.repos:17]:
```suggestion
    version: ros2-master
```

File: ros2_controllers.rolling.repos [hunk 4]
```diff
@@ -14,7 +14,7 @@ repositories:
   control_toolbox:
     type: git
     url: https://github.com/ros-controls/control_toolbox.git
-    version: ros2-master
+    version: ci/clang
```
Comment (christophfroehlich) [inline ros2_controllers.rolling.repos:17]:
```suggestion
    version: ros2-master
```

**Review by saikishor — APPROVED** [review]
LGTM in general. Some castings are new to me hahaa
If the CI is happy, I'm fine with it

**Comment by christophfroehlich** [comment]
> LGTM in general. Some castings are new to me hahaa If the CI is happy, I'm fine with it

I used now explicit iterator's `difference_type` cast instead of the `ptrdiff`(which is normally the same..)

**Pull request merged**